pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -630,6 +630,20 @@ func (ns *nodeServer) provisionDevice(device *pmdmanager.PmemDeviceInfo, fsType
 		fsType = defaultFilesystem
 	}
 
+	// Check does devicepath already contain a filesystem?
+	existingFsType, err := determineFilesystemType(device.Path)
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+	if existingFsType != "" {
+		// Is existing filesystem type same as requested?
+		if existingFsType == fsType {
+			klog.V(4).Infof("Skip mkfs as %v file system already exists on %v", existingFsType, device.Path)
+			return nil
+		} else {
```

Comment:
Go idiom is to not use else.

Reply:
deleted unnecessary else, comes in next push